File: Programmers/Level 2/32.py
from collections import deque

def solution(priorities, location):
    answer = 1
    q = deque()
    for i in range(len(priorities)):
        q.append((i, priorities[i])) # 큐에 넣는다.
    priorities.sort(reverse= True)
    
    while q:
        now = q.popleft() # 큐 앞
        if now[1] == priorities[0]:
            if now[0] == location:
                break
            priorities = priorities[1:] # 앞을 자른다
            answer += 1
        else:
            q.append(now)
    return answer

File: Programmers/Level 2/32.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    queue<pair<int, int>> q;
    priority_queue<int> pq;
        
    for(int i = 0; i < priorities.size(); i++) {
        q.push(make_pair(i, priorities[i]));
        pq.push(priorities[i]);
    }
    
    while(!pq.empty()) {
        if (pq.top() == q.front().second) {
            answer++;
            if (q.front().first == location) {
                break;
            }
            else {
                pq.pop();
                q.pop();
            }   
        }
        else {
            //answer++;
            q.push(q.front());
            q.pop(); 
        }              
    }
    
    return answer;
}
